### PR TITLE
Feat: Add run types for GW calculations

### PR DESCRIPTION
**Pull request by yanghan234**
# Added GW-related configurations and modified utilities

This PR is a dependency of a PR in the atomate2 repo [[WIP] feat: GW workflow with VASP](https://github.com/materialsproject/atomate2/pull/808). I previously submitted a PR in #1099 . However, I realized that I did not properly modified the calculation types, enums, etc. In this PR, these have been corrected.

## Contributor Checklist

- [x] Added run types and calculations types for GW calculations
- [x] Modified the utils to identify GW calculations
- [x] I have run the tests locally and they passed.

Tagging @mkhorton for awareness.

### CODE REVIEW

**Comment by tsmathis** [comment]
Thanks for resubmitting @yanghan234, your last PR got submitted right as we were updating how we generate the calc_type/run_type enums, sorry for the extra work.

Re: the empty CONTCAR, when might that apply? Is that an edge case specific to the GW workflow?

**Comment by yanghan234** [comment]
Thanks @tsmathis, yes, this is the edge case that I only oberve in the G0W0 calculation. In the G0W0 calculation, ionic steps are never updated. In addition, there is only one electronic step in such calculations, i.e. NELM = 1 or NELMGW=1. This is probably the reason that the CONTCAR is not written.

**Comment by yanghan234** [comment]
Thanks @esoteric-ephemera, please give me some time to fix them. Meanwhile, I realize that `testing/check-enums` check fails because it cannot find the branch in the forked repo under my personal account. Is this a problem with the CI setup, should I be concerned about this?

**Comment by tsmathis** [comment]
The issue with checking the enums for forks was an oversight on my part.
You don’t need to worry about that.

**Comment by yanghan234** [comment]
@esoteric-ephemera Thanks for the comments. I have updated the codes according to your comments. They are not exactly the same as what you proposed, so please take a look.

File: emmet-core/emmet/core/vasp/calculation.py [hunk 1]
```diff
@@ -748,7 +749,13 @@ def from_vasp_files(
         volumetric_files = [] if volumetric_files is None else volumetric_files
         vasprun = Vasprun(vasprun_file, **vasprun_kwargs)
         outcar = Outcar(outcar_file)
-        contcar = Poscar.from_file(contcar_file)
+        if (
+            os.path.getsize(contcar_file) == 0
```
[the remark this answers is not in the document]
Comment (yanghan234) [reply emmet-core/emmet/core/vasp/calculation.py:753]: Yes, the case is the CONTCAR written for GW but it's just empty

File: emmet-core/emmet/core/vasp/calc_types/utils.py [hunk 2]
```diff
@@ -118,6 +118,9 @@ def task_type(
     elif incar.get("IBRION", 1) == 0:
         calc_type.append("Molecular Dynamics")
 
+    elif incar.get("ALGO", "").lower() == "gw0":
```
[the remark this answers is not in the document]
Comment (yanghan234) [reply emmet-core/emmet/core/vasp/calc_types/utils.py]: This is fixed

**Comment by esoteric-ephemera** [comment]
@tsmathis while we're figuring out a secure solution for pushing to external forks, is it OK if I merge this? CI on main should fix the enum formatting as needed

**Comment by tsmathis** [comment]
Yep all good on my end

**Pull request merged**